### PR TITLE
cvpartition: fix `leaveout`method and corresponding test case

### PR DESCRIPTION
**Pull request by phychi**
The Leave-One-Out method requires that each test set contains only one sample, while the training set consists of the remaining n−1 samples.

### CODE REVIEW

**Comment by pr0m1th3as** [comment]
Thanks for the patch.
